### DISCUSSION TITLE
Error messages when policy prevents statement

### DISCUSSION THREAD

**Discussion by libovness** [supabase/supabase#1061] [Questions]
I'm currently debugging an issue where I *believe* that I'm prevented from updating rows in a table because of a policy (it doesn't seem like that should be the case but it's what I'm debugging).

In any case, the `error` for this instance returns empty array in `console.log`. Am I correct that if a `uid` does not have permission to take an action on a table it will return nothing? If I am correct, an error message would help, if possible.

**Comment by steve-chavez** [comment]
You have to use the WITH CHECK clause(only available for INSERT/UPDATE) to make UPDATE fail with an error.

Basically like:
```sql
create policy a_policy on projects
for update using (true)
with check ( col = 'your_condition');
```

**Edit**: This is a contrived example. The `using(true)` **should never be used** because it'll bypass previous SELECT policies. 

If you're adding policies through the policies UI, the policy for UPDATE will only show the USING clause(renamed to DEFINITION) by default. You can toggle "Advanced" for showing WITH CHECK.

**Reply by libovness** [reply]
thank you! 

after some more debugging it appears that my requests aren't being accepted by the policy, as it relies on the `auth.uid` equaling an `id` in the `public.users` table (this isn't the pattern I want to use but I'm simplifying temporarily in order to debug). 

what I noticed is that if I pull `user` from `Auth.useUser()` or `supabase.auth.api.getUserByCookie(req)` I get the logged in user, but `supabase.auth.currentUser` appears to be `null`. I'm assuming it's the latter which is being passed back to supabase as the `auth.uid()`. Is that assumption correct? If so, why might it be `null`?
